### PR TITLE
Live feedback

### PR DESCRIPTION
**Pull request by andreasmuelder**

### CODE REVIEW

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RelativeBendpointUtil.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package org.yakindu.base.gmf.runtime.router;
```
Comment (tkutz) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RelativeBendpointUtil.java]: Copyright & author

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+	public int sourceNeighborIndex = -1;
+	public int targetAnchorIndex = -1;
+	public int targetNeighborIndex = -1;
+	public boolean isSourceVertical = false;
+	public boolean isTargetVertical = false;
+	public int sourceSideIndex = -1;
+	public int targetSideIndex = -1;
+//	public double scale = 1d;
+//	public double tx = 0d;
+//	public double ty = 0d;
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java]: Remove commentss

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+		// determine transformations
+//		PrecisionPoint origin = new PrecisionPoint(0, 0);
+//		PrecisionPoint oneZero = new PrecisionPoint(1, 0);
+//		conn.translateToRelative(origin);
+//		conn.translateToRelative(oneZero);
+//		this.tx = origin.preciseX();
+//		this.ty = origin.preciseY();
+//		this.scale = oneZero.preciseX();
+
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java]: Remove comments

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+			MaxMoveDelta mmd = getMaxMoveDelta(boundsRel, isVertical, anchorPoint);
+			targetSideIndex = getSideIndex(boundsRel, isVertical, anchorPoint);
+			mmds[targetSideIndex].merge(mmd);
+		}
+		return mmds;
+	}
+
+//	public PrecisionPoint getTransformed(PrecisionPoint p) {
+//		return new PrecisionPoint((p.preciseX() * scale) + tx, (p.preciseY() * scale) + ty);
+//	}
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/ConnData.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+		}
+
+		// search from start so that removals can happen without having to adjust
+		// indices
+		for (int i = points.size() - 1; i >= (minLoopSize - 1); i--) {
+			final Line rearLine = new Line(geom.toPP(points.get(i - 1)), geom.toPP(points.get(i)));
+			for (int j = 0; j <= ((i - minLoopSize) + 1); j++) {
+				final Line frontLine = new Line(geom.toPP(points.get(j)), geom.toPP(points.get(j + 1)));
+				final PrecisionPoint poi = geom.getIntersection(rearLine, frontLine);
+//				System.out.println("intersect " + rearLine + " with " + frontLine + " yields " + poi);
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+		}
+		initDrag(originalAbs, pureSource, true, false);
+		initDrag(originalAbs, pureTarget, false, true);
+		initDrag(originalAbs, reflexive, true, true);
+	}
+
+	private void initDrag(Rectangle originalAbc, List<Connection> connections, boolean isSource, boolean isTarget) {
+		for (Connection connection : connections) {
+			// disable repaint so that first drag does not let connection jump
+//			((TransitionFigure) connection).disableRepaint();
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+			double localDx = bnrel.preciseX() - brel.preciseX();
+			double localDy = bnrel.preciseY() - brel.preciseY();
+			double localDw = bnrel.preciseWidth() - brel.preciseWidth();
+			double localDh = bnrel.preciseHeight() - brel.preciseHeight();
+
+			// drag anchored segment if necessary
+			List<PrecisionPoint> pointsCopy = dragAnchoredSegments(cd, localDx, localDy, localDw, localDh);
+
+//			cd.printVisualPoints();
+//			cd.printPoints(pointsCopy);
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+
+			List<RelativeBendpoint> constraint = createConstraint(cd.conn, pointsCopy);
+			ConnectionRouter router = cd.conn.getConnectionRouter();
+
+			// disable repaint to prevent flickering when restoring original constraint
+			// (hard to reproduce but happens)
+//			((TransitionFigure) cd.conn).disableRepaint();
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+			Rectangle sourceBox = isSelfAssoc ? targetBox : getBounds(cd.conn, sourceOwner);
+
+			cutOffEnd(list, targetBox, isSelfAssoc);
+			cutOffStart(list, sourceBox, isSelfAssoc);
+
+			cutOutLoops(list);
+
+			List<RelativeBendpoint> updatedConstraint = createConstraint(cd.conn, list);
+
+//			((TransitionFigure) cd.conn).enableRepaint();
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+			router.setConstraint(cd.conn, updatedConstraint);
+			router.route(cd.conn);
+
+//			System.out.println("after all");
+//			cd.printVisualPoints();
+//			System.out.println();
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.gmf.runtime/src/org/yakindu/base/gmf/runtime/router/RubberBandRoutingSupport.java]: Remove

File: plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java [hunk 11]
```diff
@@ -54,35 +63,54 @@
...
-		this.newPointList = new PointList(newPointList.size());
-		for (int i = 0; i < newPointList.size(); i++) {
-			this.newPointList.addPoint(newPointList.getPoint(i));
+	private void computeGMFLabelOffset(LabelEditPart labelEditPartToUpdate, ConnectionEditPart connectionEditPart) {
+		Point newLabelOffset = null;
+		Node labelNodeToUpdate = (Node) labelEditPartToUpdate.getModel();
+		if (connectionEditPart.getModel() instanceof Edge) {
+			PointList oldBendpoints = oldBendPointsList;
+			if (oldBendpoints == null) {
+//				System.out.println("read current points");
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java]: Remove

File: plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java [hunk 12]
```diff
@@ -54,35 +63,54 @@
...
+				oldBendpoints = connectionEditPart.getConnectionFigure().getPoints();
+			}
+			boolean isEdgeWithObliqueRoutingStyle = isEdgeWithObliqueRoutingStyle(connectionEditPart);
+			LayoutConstraint layoutConstraint = labelNodeToUpdate.getLayoutConstraint();
+			if (layoutConstraint instanceof Location) {
+				Location point = (Location) layoutConstraint;
+				newLabelOffset = new EdgeLabelQuery(oldBendpoints, newPointList, isEdgeWithObliqueRoutingStyle,
+						new Point(point.getX(), point.getY()), labelEditPartToUpdate.getFigure().getSize(),
+						labelEditPartToUpdate.getKeyPoint(), false).calculateGMFLabelOffset();
+//				System.out.println("queried label offset = " + newLabelOffset);
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java]: Remove

File: plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java [hunk 13]
```diff
@@ -54,35 +63,54 @@
...
+				if ((newLabelOffset.x == 0) && (newLabelOffset.y == 0)) {
+					newLabelOffset = null;
+				}
+			}
+		}
+
+		if (newLabelOffset != null) {
+//			System.out.println("store label offset");
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java]: Remove

File: plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java [hunk 14]
```diff
@@ -176,6 +175,7 @@ public static boolean isEdgeWithObliqueRoutingStyle(org.eclipse.gef.ConnectionEd
 	 * this method has no effect.
 	 */
 	public void updateGMFLabelsOffset() {
+//		System.out.println("update label offsets");
```
Comment (BeckmaR) [inline plugins/org.yakindu.base.xtext.utils.gmf/src/org/yakindu/base/xtext/utils/gmf/commands/SetLabelsOffsetOperation.java]: Remove

**Pull request merged**